app/models/taxon_lineage.rb

```diff
@@ -46,7 +46,9 @@ def self.fill_lineage_details(tax_map, pipeline_run_id)
 
     # Get created_at date for our TaxonCount entries. Same for all TaxonCounts
     # in a PipelineRun.
-    valid_date = PipelineRun.find(pipeline_run_id).taxon_counts.last.created_at
+    # TODO: change this to get the created_at date from the smaller pipeline_run table.
+    #       And/or move the lineage selection mechanism into alignment_config.
+    valid_date = TaxonCount.where(pipeline_run_id: pipeline_run_id).select(:created_at).last.created_at
```

Comment:
I thought you are going to change
PipelineRun.find(pipeline_run_id).created_at?

Reply:
Yes, I thought so too. But there are other places in the code where it uses the taxon_count date, so I left it in the interest of consistency. It will be a bigger undertaking to refactor all the places to use pipeline_run date -- want to do it properly (probably using alignment_config). That's why I added it as a TODO. Please let me know what you think.